pkg/kube/snapshot/snapshot_alpha.go

```diff
@@ -113,18 +114,61 @@ func (sna *SnapshotAlpha) Create(ctx context.Context, name, namespace, pvcName s
 }
 
 // Get will return the VolumeSnapshot in the 'namespace' with given 'name'.
-func (sna *SnapshotAlpha) Get(ctx context.Context, name, namespace string) (*v1alpha1.VolumeSnapshot, error) {
+func (sna *SnapshotAlpha) Get(ctx context.Context, name, namespace string) (*v1.VolumeSnapshot, error) {
 	us, err := sna.dynCli.Resource(v1alpha1.VolSnapGVR).Namespace(namespace).Get(ctx, name, metav1.GetOptions{})
 	if err != nil {
 		return nil, err
 	}
 
 	vs := &v1alpha1.VolumeSnapshot{}
-	return vs, TransformUnstructured(us, vs)
+	if err := TransformUnstructured(us, vs); err != nil {
+		return nil, err
+	}
+
+	// Populate v1.VolumeSnapshot object from v1alpha1.VolumeSnapshot
+	vsRet := v1.VolumeSnapshot{}
+	meta := vs.ObjectMeta.DeepCopy()
+	if meta == nil {
+		return nil, fmt.Errorf("Invalid VolumeSnapshotObject: ObjectMeta is nil")
+	}
+	vsRet.ObjectMeta = *meta
+
+	if vs.Spec.Source != nil && vs.Spec.Source.Kind == "PersistentVolumeClaim" {
+		vsRet.Spec.Source.PersistentVolumeClaimName = &vs.Spec.Source.Name
+	}
+	if vs.Spec.VolumeSnapshotClassName != "" {
+		vsRet.Spec.VolumeSnapshotClassName = &vs.Spec.VolumeSnapshotClassName
+	}
+	if vs.Spec.SnapshotContentName != "" {
+		vsRet.Spec.Source.VolumeSnapshotContentName = &vs.Spec.SnapshotContentName
+	}
+
+	if vs.Status == (v1alpha1.VolumeSnapshotStatus{}) {
```

Comment:
Do you need to check if its nil? Or will it always be loaded with an empty object?

Reply:
We do not always get an empty object, if we do, we return from here